ddl/index_cop.go

```diff
@@ -213,26 +187,21 @@ func (c *copReqSenderPool) adjustSize(n int) {
 	}
 }
 
-func (c *copReqSenderPool) close() {
-	logutil.BgLogger().Info("[ddl-ingest] close cop-request sender pool", zap.Int("results not handled", len(c.results.Keys())))
+func (c *copReqSenderPool) close(force bool) {
+	if c.closed {
+		return
+	}
+	logutil.BgLogger().Info("[ddl-ingest] close cop-request sender pool")
 	close(c.tasksCh)
-	for _, w := range c.senders {
-		w.cancel()
+	if force {
```

Comment:
I saw int backfilling.go:870 normal process will close(false),  and in backfilling.go:811 -> defer scheduler.close(true). if this will cause there is no chance to close closeBackfillWorkers(b.workers)?

Reply:
Yes. Maybe `closeBackfillWorkers` should be renamed into `cancelBackfillWorkers`?

All it does is notify workers to shutdown.